### ISSUE TITLE
There is a bug when entering Chinese on the terminal, and the content is not displayed

### ISSUE THREAD

**Issue by h3lp404** [warpdotdev/Warp#4700]
### Describe the bug

There is a bug when entering Chinese on the terminal, and the content is not displayed

### To reproduce

When you enter Chinese content in the terminal or edit content, the content will not be displayed

### Operating system

MacOS

### Operating system and version

14.4.1

### Current Warp version

v0.2024.04.02.08.02.stable_01

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @h3lp404 thanks for letting us know. This may be due to the font which you're using, try make sure you have a font with Chinese characters set in your settings > appearance > fonts.

**Comment by h3lp404** [comment]
> Hi @h3lp404 thanks for letting us know. This may be due to the font which you're using, try to make sure you have a font with Chinese characters set in your settings > appearance > fonts. This may be due to the font you are using, please try to make sure you have a font with Chinese characters set in your Settings> Appearance> Fonts.


> Hi @h3lp404 thanks for letting us know. This may be due to the font which you're using, try make sure you have a font with Chinese characters set in your settings > appearance > fonts.


No, not fonts setting ,I use hack font, the content can be displayed, but when editing(ex:vim test.txt), the front text content will not be displayed, and the front content will not be displayed until the content appears later.

**Comment by h3lp404** [comment]
> Hi @h3lp404 thanks for letting us know. This may be due to the font which you're using, try make sure you have a font with Chinese characters set in your settings > appearance > fonts.

like this:
![Longshot-20240408104235](https://github.com/warpdotdev/Warp/assets/112472696/9e8bf268-612e-490f-b2a9-7a62f3a2cb80)

**Comment by eliefly** [comment]
I encountered a similar issue on Windows.

  Environment:
  - OS: Windows 11 Pro, build 26200
  - IME: Microsoft Pinyin
  - Warp version: v0.2026.04.08.08.36.stable_05
  - Shell: PowerShell

  Behavior:
  - Chinese IME preedit/composition text is not visible while typing
  - This happens in both a normal PowerShell prompt and inside Codex CLI in Warp

  I tested the following and the issue still persists:
  - turned off Warp AI features and restarted Warp
  - switched to Classic Input / Shell (PS1)
  - disabled Input Auto-detection
  - enabled "Use previous version of Microsoft Pinyin"

  For comparison, the same Microsoft Pinyin IME works correctly in Windows Terminal on the same machine, where the preedit text (for example, `nihao`)  is visible while typing.
I attached a screenshot showing that Windows Terminal displays the IME preedit text correctly, which suggests the issue is specific to Warp rather than a general Windows IME problem.

<img width="404" height="185" alt="Image" src="https://github.com/user-attachments/assets/2c044edf-1431-46fb-af41-978285121b8d" />
